nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -313,8 +328,18 @@ public void process(InputStream in) throws IOException {
                                 if (conflictResponse.equals(APPEND_RESOLUTION_AV.getValue()) && destinationExists) {
                                     fos = hdfs.append(copyFile, bufferSize);
                                 } else {
-                                    fos = hdfs.create(tempCopyFile, true, bufferSize, replication, blockSize);
-                                }
+                                  final EnumSet<CreateFlag> cflags = EnumSet.of(CreateFlag.CREATE, CreateFlag.OVERWRITE);
+
+                                  final Boolean ignoreLocality = context.getProperty(IGNORE_LOCALITY).asBoolean();
+                                  if (ignoreLocality.booleanValue()) {
```

Comment:
Minor: Could use simply `ignoreLocality` instead of `ignoreLocality.booleanValue()`.